.pre-commit-config.yaml

```diff
@@ -82,6 +82,15 @@ repos:
         always_run: true
         entry: pytest --doctest-modules src/pudl test/unit -m "not slow"
 
+      - id: conda-lock
+        name: conda-lock
+        stages: [commit]
+        language: system
+        verbose: false
+        files: "pyproject.toml"
+        always_run: false
+        entry: make conda-clean conda-lock.yml
```

Comment:
Surprisingly straightforward to get the behavior I wanted... a little suspicious but I'll take it.

Reply:
I made a trivial change to my local `pyproject.toml` and tried to commit and it regenerated the lockfiles, refused to commit, and then I added them to the staging area, and tried to commit again, and it regenerated them again with no changes and the commit went through so, it seems like it works.

I guess if someone makes a change to `pyproject.toml` that is NOT related to dependencies, after the first run through the pre-commit hooks if everything else passes they can do a `--no-verify` to avoid unnecessarily relocking all the dependencies.